game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -140,7 +140,7 @@ public void gameDataChanged(final Change change) {
             tileManager.resetTiles(gameData, uiContext.getMapData());
             SwingUtilities.invokeLater(
                 () -> {
-                  recreateTiles(getData(), uiContext);
+                  recreateTiles(gameData, uiContext);
```

Comment:
This is actually a thing I wonder which "style" to prefer, maybe something we should mention in our code guidelines

Reply:
IIRC, there is a sonar lint violation about it. Using a method to access local data is an anti-pattern IMO. The method can be overriden, which then can cause unexpected breaks in methods that were previously working. Even if there are no breakages, it becomes very non-intuitive looking at code how it would work as the methods can behave differently depending on the instance type. This removes the ability to reason about a class by only looking at its code, you have to understand the usage history and system state to know exactly how the class will behave. Hence, anti-pattern, it makes it hard to reason how this class will work and breaks encapsulation.